### PR TITLE
Add --with-boot-jdk and other improvements to Linux build instructions

### LINKED ISSUE #9375: Build instructions don't generally mention --with-boot-jdk=
The docker script https://github.com/eclipse/openj9/pull/9371 provides all the prerequisite boot jdks, but the build doc doesn't generally mention using the configure option `--with-boot-jdk=`. We should add this for all platforms, it's only mentioned in the Windows build instructions atm.

### PR DESCRIPTION
**Pull request by pshipton**
Also modify mkdocker.sh to stop installing jdk13 since the instructions
no longer refer to this version.

Closes #9375

### CODE REVIEW

File: doc/build-instructions/Build_Instructions_V8.md [hunk 1]
```diff
@@ -104,14 +104,37 @@ apt-get update \
    ...
 ```
 
-2. Download and setup **freemarker.jar** into a directory. The example commands use `/root` to be consistent with the Docker instructions. If you aren't using Docker, you probably want to store the **freemarker.jar** in your home directory.
+2. This build uses the same gcc and g++ compiler levels as OpenJDK, which might be
```
Comment (pshipton) [inline doc/build-instructions/Build_Instructions_V8.md:107]: @keithc-ca you removed this section from the non-docker instructions in #9371 and I'm not entirely sure why.

Comment (keithc-ca) [reply doc/build-instructions/Build_Instructions_V8.md:107]: That change went with the advice to use cmake; which apparently was only partly reverted.

File: doc/build-instructions/Build_Instructions_V11.md [hunk 2]
```diff
@@ -171,10 +171,12 @@ bash get_source.sh
 :penguin:
 When you have all the source files that you need, run the configure script, which detects how to build in the current build environment.
 ```
-bash configure --with-freemarker-jar=/<my_home_dir>/freemarker.jar
+bash configure --with-freemarker-jar=/<my_home_dir>/freemarker.jar --with-boot-jdk=/usr/lib/adoptojdk-java-11
 ```
 :warning: You must give an absolute path to freemarker.jar
 
+:warning: The path in the example --with-boot-jdk= option is appropriate for the Docker installation, this option is not required if you followed the instructions to download and setup the boot JDK.
```
Comment (keithc-ca) [inline doc/build-instructions/Build_Instructions_V11.md:178]: I would reword this: the option is required if `JAVA_HOME` isn't set or doesn't refer to a suitable boot jdk. If you're building multiple jdk versions, you probably can't satisfy them all with `JAVA_HOME`. I think we should advise use of `--with-boot-jdk` in general.

File: doc/build-instructions/Build_Instructions_V14.md [hunk 3]
```diff
@@ -169,10 +169,12 @@ bash get_source.sh
 :penguin:
 When you have all the source files that you need, run the configure script, which detects how to build in the current build environment.
 ```
-bash configure --with-freemarker-jar=/<my_home_dir>/freemarker.jar --with-boot-jdk=<path_to_boot_JDK13>
+bash configure --with-freemarker-jar=/<my_home_dir>/freemarker.jar --with-boot-jdk=/usr/lib/adoptojdk-java-14
 ```
 :warning: You must give an absolute path to freemarker.jar
 
+:warning: The path in the example --with-boot-jdk= option is appropriate for the Docker installation, this option is not required if you followed the instructions to download and setup the boot JDK.
```
Comment (keithc-ca) [inline doc/build-instructions/Build_Instructions_V14.md:176]: Same comment as above.

File: doc/build-instructions/Build_Instructions_V8.md [hunk 4]
```diff
@@ -133,10 +155,12 @@ bash get_source.sh
 :penguin:
 When you have all the source files that you need, run the configure script, which detects how to build in the current build environment.
 ```
-bash configure --with-freemarker-jar=/root/freemarker.jar
+bash configure --with-freemarker-jar=/root/freemarker.jar --with-boot-jdk=/usr/lib/adoptojdk-java-80
 ```
 :warning: You must give an absolute path to freemarker.jar
 
+:warning: The path in the example --with-boot-jdk= option is appropriate for the Docker installation, this option is not required if you followed the instructions to download and setup the boot JDK.
```
Comment (keithc-ca) [inline doc/build-instructions/Build_Instructions_V8.md:162]: And again.

File: doc/build-instructions/Build_Instructions_V11.md (hunk 2 above)
Comment (pshipton) [reply doc/build-instructions/Build_Instructions_V11.md]: I've updated, but if you still want changes we can figure it out on Slack.

**Comment by keithc-ca** [comment]
Can you fix the links to mkdocker.sh while you're here, please?
References to
```
https://github.com/eclipse/openj9/blob/master/buildenv/mkdocker.sh
```
should be
```
https://github.com/eclipse/openj9/blob/master/buildenv/docker/mkdocker.sh
```

File: doc/build-instructions/Build_Instructions_V8.md [hunk 5]
```diff
@@ -104,14 +104,36 @@ apt-get update \
...
-2. Download and setup **freemarker.jar** into a directory. The example commands use `/root` to be consistent with the Docker instructions. If you aren't using Docker, you probably want to store the **freemarker.jar** in your home directory.
+2. This build uses the same gcc and g++ compiler levels as OpenJDK, which might be
+backlevel compared with the versions you use on your system. Create links for
+the compilers with the following commands:
+```
+ln -s g++ /usr/bin/c++
+ln -s g++-7 /usr/bin/g++
+ln -s gcc /usr/bin/cc
+ln -s gcc-7 /usr/bin/gcc
+```
```
Comment (keithc-ca) [inline doc/build-instructions/Build_Instructions_V8.md:115]: If we're going to discuss this, I think it should explain how to get `gcc-7` and `g++-7` that are the targets of those links.

**Comment by keithc-ca** [comment]
It would be helpful to note that the reference repo in docker containers will speed the clone step and reduce the additional disk space needed:
```
git clone --reference /home/jenkins/openjdk_cache https://github.com/ibmruntimes/openj9-openjdk-jdk8.git
```

File: doc/build-instructions/Build_Instructions_V8.md (hunk 5 above)
Comment (pshipton) [reply doc/build-instructions/Build_Instructions_V8.md:115]: This should be covered by the following, as the apt-get list includes `gcc-7 and g++-7`: `1. Install the list of dependencies that can be obtained with the `apt-get` command from the following section of the Dockerfile`.

**Comment by pshipton** [comment]
Fixed the mkdocker.sh link. Perhaps the other comments should be addressed in a separate PR?

**Comment by keithc-ca** [comment]
> Fixed the mkdocker.sh link. Perhaps the other comments should be addressed in a separate PR?

This add a mention of `gcc-7`, I think it's reasonable to expect that it explain where to get that.

I'm fine deferring the reference cache issue.

**Comment by pshipton** [comment]
Technically I didn't add gcc-7, I restored what was accidentally removed. Anyway, I've pushed a new commit for you to consider. Once we finalize the working, I can update V11 and V14 similarly and squash.

File: doc/build-instructions/Build_Instructions_V8.md (hunk 5 above)
Comment (keithc-ca) [reply doc/build-instructions/Build_Instructions_V8.md]: The section that was removed referred to gcc-4.8; the current requirement is for gcc-7.5. If you follow the steps listed in `mkdocker.sh --print`, you'll get gcc-7.5 and symbolic links are not required. I think this section should just stay removed.

Comment (pshipton) [reply doc/build-instructions/Build_Instructions_V8.md]: What about this part? `ENV CC=gcc-7 CXX=g++-7`
Are these exports required to use gcc 7 in the build in place of whatever else the user has installed?

Comment (pshipton) [reply doc/build-instructions/Build_Instructions_V8.md]: As a test, on Ubuntu 20 I did `apt install gcc` which installed gcc 9. Followed by `apt install gcc-7` but this doesn't replace the gcc 9 default you get when running `gcc --version`.

Comment (keithc-ca) [reply doc/build-instructions/Build_Instructions_V8.md]: If you use `apt install g++-7` then those environment variables will select that version (independent of whether you have other versions installed).

If you use `apt install g++` you get gcc-9 which worked last time I tried; those environment variables are not be needed in that case. On Ubuntu 18.04, you'll get version 7, so neither the symlinks nor the environment variables are necessary.

Comment (pshipton) [reply doc/build-instructions/Build_Instructions_V8.md]: I've pushed a new commit.

Comment (pshipton) [reply doc/build-instructions/Build_Instructions_V8.md]: I think it's better to tell the user to explicitly set the environment variables and use gcc 7. This is the compiler version we test and recommend. Otherwise we don't know what the use might have installed. Ub 16 installs gcc 5 by default which I don't think works.

File: doc/build-instructions/Build_Instructions_V8.md [hunk 6]
```diff
@@ -99,19 +99,36 @@ bash mkdocker.sh --tag=openj9 --dist=ubuntu --version=16.04 --print
...
+4. Download and setup the boot JDK using the latest AdoptOpenJDK v8 build.
+```
+cd /<my_home_dir>
+wget -O bootjdk8.tar.gz "https://api.adoptopenjdk.net/v3/binary/latest/8/ga/linux/x64/jdk/openj9/normal/adoptopenjdk"
+tar -xzf bootjdk8.tar.gz
+rm -f bootjdk8.tar.gz
+mv $(ls | grep -i jdk) bootjdk8
+
+export JAVA_HOME="/<my_home_dir>/bootjdk8"
+export PATH="${JAVA_HOME}/bin:${PATH}"
```
Comment (keithc-ca) [inline doc/build-instructions/Build_Instructions_V8.md:129]: This can be removed given we're recommending use of `--with-boot-jdk`.

Comment (pshipton) [reply doc/build-instructions/Build_Instructions_V8.md]: Pushed another update.

File: buildenv/docker/mkdocker.sh [hunk 7]
```diff
@@ -569,7 +569,7 @@ bootjdk_url() {
 }
 
 install_bootjdks() {
-  local versions="8 11 13 14"
+  local versions="8 11 14"
```
Comment (keithc-ca) [inline buildenv/docker/mkdocker.sh:572]: `defaults.yml` still says:
```
  boot_jdk:
    14: '/usr/lib/jvm/adoptojdk-java-13'
```
So I don't think it's yet time to remove jdk13 here.

**Comment by keithc-ca** [comment]
11 and 14 should also be updated like 8 to replace references to files like `buildenv/docker/jdk11/x86_64/ubuntu16/Dockerfile` with instructions for using the new `mkdocker.sh` script.

File: buildenv/docker/mkdocker.sh (hunk 7 above)
Comment (pshipton) [reply buildenv/docker/mkdocker.sh:572]: I'll open a PR to change it. Just because OpenJ9 builds still use 13 doesn't mean we can't tell users to use 14.

**Comment by pshipton** [comment]
> 11 and 14 should also be updated like 8

I thought you had indicated you would do it, but I will make the changes.

File: buildenv/docker/mkdocker.sh (hunk 7 above)
Comment (keithc-ca) [reply buildenv/docker/mkdocker.sh:572]: As long as OpenJ9 builds (that could use the docker image) use 13, it should stay.
I wasn't objecting to the other file that advises using 14.

Comment (pshipton) [reply buildenv/docker/mkdocker.sh:572]: I don't think there is any relation between defaults.yml and running in docker atm. 
Created https://github.com/eclipse/openj9/pull/9422

**Comment by pshipton** [comment]
Added a commit to update 11 and 14 like 8.
